apps/api-extractor/src/analyzer/TypeScriptInternals.ts

```diff
@@ -83,12 +83,26 @@ export class TypeScriptInternals {
    */
   public static getResolvedModule(
     sourceFile: ts.SourceFile,
-    moduleNameText: string
+    moduleNameText: string,
+    mode: ts.StringLiteralLike | undefined
   ): ts.ResolvedModuleFull | undefined {
     // Compiler internal:
     // https://github.com/microsoft/TypeScript/blob/v3.2.2/src/compiler/utilities.ts#L218
 
-    return (ts as any).getResolvedModule(sourceFile, moduleNameText);
+    return (ts as any).getResolvedModule(sourceFile, moduleNameText, mode);
+  }
+
+  /**
+   * Gets the mode required for module resolution required with the addition of Node16/nodenext
+   */
+  public static getModeForUsageLocation(
+    file: { impliedNodeFormat?: ts.SourceFile['impliedNodeFormat'] },
+    usage: ts.StringLiteralLike | undefined
+  ) {
+    // Compiler internal:
+    // https://github.com/microsoft/TypeScript/blob/v4.7.2/src/compiler/program.ts#L568
+
+    return (ts as any).getModeForUsageLocation(file, usage);
```

Comment:
This function might not exist in older versions. It should just return `undefined` if it doesn't have a value.

```suggestion
    return (ts as any).getModeForUsageLocation?.(file, usage);
```